service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapper.java

```diff
@@ -57,11 +64,15 @@ private EhrExtractTemplateParameters prepareEhrFhirExtractParamsFromFhirBundle(
         ehrExtractTemplateParameters.setToOdsCode(getGpcStructuredTaskDefinition.getToOdsCode());
         ehrExtractTemplateParameters.setFromOdsCode(getGpcStructuredTaskDefinition.getFromOdsCode());
         ehrExtractTemplateParameters.setAvailabilityTime(DATE_TIME_FORMATTER.format(timestampService.now()
-            .atOffset(ZoneOffset.UTC)));
+            .atZone(ZoneId.of(UK_ZONE_ID))));
 
         extractPatientFromBundle(bundle)
             .orElseThrow(() -> new FhirValidationException("Missing patient resource in Fhir Bundle."));
 
+        var encounters = EncounterExtractor.extractEncountersByReferences(bundle.getEntry(),
```

Comment:
For me it would be easier understand the business logic that' behind list's mapping, if we would have just one method  extractEncountersFromConsultationsList(bundle.getEntry()) , instead of two.

Reply:
I was thinking about this. However, notice that EncounterExtractor is a class which only extracts encounters from bundle, while the mapper class maps provided encounter to an xml format. I think we should think here about single responsibility principle.

Reply:
EncounterExtractor has method extractEncountersByReferences that returns list of Encounters.

Reply:
I was thinking about moving logic of extractEncountersByReferences and extractEncountersByReferences into one, while they are both from the same class.

Reply:
Ok